lib/docassemble/docassemble_api_interface.js

```diff
@@ -0,0 +1,377 @@
+*    there's nothing we can do about that.
+*/
+
+let da_i = {};
+module.exports = da_i;
+
+
+let is_timeout_error = function( error ) {
+  /** Return whether the error is an axios timeout error.
+  * See:
+  *    Check on https://github.com/axios/axios/issues/1543#issuecomment-558166483 (2019)
+  *    and https://github.com/axios/axios/issues/1174#issuecomment-349014752 (2017)
+  */
+  if ( error.message && error.message.includes( `timeout` ) ) {
+    return true;
+  }
+  return false;
+};
+
+
+// node -e 'require("./lib/github_integration/docassemble_api_interface.js").throw_an_error_if_server_is_not_responding()'
+da_i.throw_an_error_if_server_is_not_responding = async function ( options ) {
+  /** Make a request to the server to see if it's responding.
+  *    It doesn't matter what the request is, as long as it doesn't
+  *    cause a server restart.
+  *    This is currently meant for puppeteer tests, not for here.
+  *
+  *    If we do add a loop to poll the server, we can add it in here:
+  * let { time_per_loop, max_loops } = options || { time_per_loop: 5 * 1000, max_loops: 15 };
+  */
+
+  // Defaults
+  let { timeout } = options || { timeout: 5 * 1000 };
+  log.info({ pre: `Waiting for the docassemble server to respond. Will wait for ${ timeout/1000 } seconds at most.` });
+  
+  try {
+    let start_time = Date.now()
+    // Axios + server will poll until server is ready or timeout is over
+    let response = await _da.get_dev_id( timeout );
+    let end_time = Date.now()
+    if ( response && response.data && response.data.id ) {
+      log.info({ pre: `The docassemble server responded after ${ (end_time - start_time)/1000 } seconds.` });
+      return false;
+    }
+
+  } catch ( error ) {
+    // Timeout means the server is busy.
+    if ( is_timeout_error( error ) ) {
+      // Anything that's waiting for the server to not be busy does not want
+      // to move forward without a free server.
+      log.info({ pre: `The docassemble server was still busy after ${ timeout/1000 } seconds.` });
+    } else {
+      // Not sure what to do for other errors. It could be a server
+      // error, like an 'Access Denied' error, which should be handled.
+      log.warn({ pre: `Tried to check if the docassemble server was busy, but ran into an unexpected `
+        + `error. The error didn't indicate a busy server, though.`
+      });
+    }
+
+    throw error;
+  }
+};  // Ends da_i.throw_an_error_if_server_is_not_responding()
+
+
+// node -e 'require("./lib/github_integration/docassemble_api_interface.js").get_dev_id()'
+da_i.get_dev_id = async function ( dev_id_options ) {
+  /** For consistency, wrap _da.get_dev_id().
+  *    See https://docassemble.org/docs/api.html#user.
+  *    Should we throw the 'Access Denied' error in here instead? */
+
+  // Default timeouts
+  let { timeout } = dev_id_options ||  { timeout: 75 * 1000 };
+  return await _da.get_dev_id( timeout );
+};  // Ends da_i.get_dev_id();
+
```

Comment:
Something to think about:
* We could create two functions: one to check if the server is active, and another to actually return the developer ID.  The first one will just validate that a develop ID is true and return true if the value is returned correctly.  The other will actually return the developer ID.